### PR TITLE
Fix truss push output for TGI.

### PR DESCRIPTION
**Pull request by squidarth**
# Summary

Fixes a couple issues with the `truss push` path:
* It currently assumes that if the user did not pass `--publish` that they deployed a draft model. With TGI/vLLM, this is not true, since they are always non-draft, despite what flags are passed. This fixes that
* There were numerous type errors that we had ignored. I changed the API to resolve these, which I hope is an improvement 
* Removed the "Open  logs in new tab" thing that I've been finding annoying (other users have noted this too) when deploying models

# Testing

## Deploying a Draft Truss

```
$ poetry run truss push --model-name "test-100" examples/truss-public-gh-repo-test/
? 🎮 Which remote do you want to connect to? sid-prod
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
✨ Model test-100 was successfully pushed ✨

|---------------------------------------------------------------------------------------|
| Your model has been deployed as a draft. Draft models allow you to                    |
| iterate quickly during the deployment process.                                        |
|                                                                                       |
| When you are ready to publish your deployed model as a new version,                   |
| pass `--publish` to the `truss push` command. To monitor changes to your model and    |
| rapidly iterate, run the `truss watch` command.                                       |
|                                                                                       |
|---------------------------------------------------------------------------------------|

🪵  View logs for your deployment at https://app.baseten.co/models/1Py97GP/versions/w5dmnm3/logs
```
## Deploying a Prod truss

```
 $ poetry run truss push --model-name "test-100" examples/truss-public-gh-repo
-test/ --publish
? 🎮 Which remote do you want to connect to? sid-prod
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
✨ Model test-100 was successfully pushed ✨
🪵  View logs for your deployment at https://app.baseten.co/models/1Py97GP/versions/w678n2q/logs
```

## Deploying a TGI Truss
```
$ poetry run truss push examples/tgi/
? 🎮 Which remote do you want to connect to? sid-prod
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
✨ Model Falcon-TGI-2 was successfully pushed ✨
🪵  View logs for your deployment at https://app.baseten.co/models/gPGpzKq/versions/qelm853/logs
```

### CODE REVIEW

File: truss/remote/truss_remote.py [hunk 1]
```diff
@@ -142,6 +142,13 @@ def authenticate(self) -> dict:
         """
         return {}
```
Comment (squidarth) [inline truss/remote/truss_remote.py:144]: All the new types in this file are defined to satisfy the type checks. Can evolve the next time we have a truss remote

Comment (aspctu) [reply truss/remote/truss_remote.py:144]: Can you satisfy type checks by adding these methods to the Baseten remote subclass? The `TrussRemote` class is ideally a very minimal abstract class that provides an interface for all remotes that accept trusses. Adding log / development version functionality to the TrussRemote class breaks this as other remotes, such as SageMaker, may not have such abstractions.

Comment (squidarth) [reply truss/remote/truss_remote.py:144]: Short answer is -- no

The truss "cli.py" push / watch functions expect a generic remote object, not a Baseten remote object. So we can only call methods in those functions that exist on the `TrussRemote` class. Our options are either:
* Change `truss push` and `truss watch` to only work with Baseten remotes (which means these commands won't work with sagemaker & other remotes)
* Remove the viewing logs and truss watch functionality from the CLI

I think neither of these options are great. Can we go w/ this for now, and for other remotes simply raise `NotImplemented`? I think it's hard to have these conversations without knowing what the other remotes would be like / what `truss push` looks like in that world.

**Review by aspctu — APPROVED** [review]

File: truss/remote/truss_remote.py (hunk 1 above)
Comment (aspctu) [reply truss/remote/truss_remote.py:144]: Approved, we can revisit when a new implementation is added.

**Pull request merged**